### PR TITLE
Stop leaving temp files behind (finishes up on #25)

### PR DESCRIPTION
**Pull request by hartwork**
Based on (a rebased version of) pull request #25 by @fredden

### CODE REVIEW

**Comment by esc** [comment]
I wonder if there is any way to test this?

**Comment by hartwork** [comment]
> I wonder if there is any way to test this?

If you like the general direction I can add tests on top. I have an idea how to do that in Cram/Scruf. It would be good to have a decision on #50 first so I'm sure I work into the right direction.

**Comment by hartwork** [comment]
Rebased onto latest `master` (which includes #50) and added a simple test now.

**Review by esc — CHANGES_REQUESTED** [review]
I reviewed it by testing it on my local machine. I used `--debug` to trace that the file is created and that the wait time is read from the settings and applied correctly. Also in the case of `1`. I can confirm the file is deleted and also I can confirm that the viewer will not find the file if it is deleted too early. I also manually examined the test case and can confirm that it appears to work as intended. I left a few minor language nitpicks -- but the PR is ready to merge once the nitpicks have been addressed. 

`debug:   Value for 'wait' found in 'conf file', overrides setting '2.0' from 'defaults': '1.0'` <--- this line made me smile in the output.

File: test.scf [hunk 1]
```diff
@@ -110,6 +111,12 @@ provide filename and viewer and format
     $ rm file.png
     $ ls
 
+check for deletion of temp files
+    $ ls -1 "$(git-big-picture -d -v true | fgrep 'temp file' | tee /dev/stderr | grep -Eo "/[^']+" | uniq)"
```
[the remark this answers is not in the document]
Comment (hartwork) [reply test.scf:115]: Sure! Pattern `/[^']+` is meant to match an absolute filename — some string starting with a slash —, the tempfile. It's enclosed in the output by single ticks left and right so I used `[^']` — anything but a single tick — to capture the full filename but not the closing tick. grep option `-o` ("only") drops the parts left and right of the match from any matching lines and `-E` ("extended") is to keep the regex more readable and not have `+` need to be `\+` (but it's not too important).

File: README.rst [hunk 2]
```diff
@@ -189,6 +189,12 @@ Usage
         -o FILE, --outfile=FILE
                             write image to specified file
         -O, --no-outfile    disable writing image to file
+        -w SECONDS, --wait=SECONDS
+                            wait for SECONDS seconds before deleting the temporary
+                            file that is opened using the viewer command (default:
+                            2.0 seconds); this helps e.g. with viewer commands
+                            that tell other running processes to open that file on
+                            their behalf, to then shut down themselves
```
[the remark this answers is not in the document]
Comment (hartwork) [reply README.rst]: Sure! Fixed x2

File: git-big-picture [hunk 3]
```diff
@@ -954,17 +980,30 @@ def main(opts, args):
...
         write_to_file(output_settings[OUT_FILE], dot_output)
         if output_settings[VIEWER]:
             debug("Will now open file in viewer: '%s'" %
                     output_settings[VIEWER])
+            if temporary_file is not None:
+                wait_until = time.time() + max(0, output_settings[WAIT_SECONDS])
             show_in_viewer(output_settings[OUT_FILE], output_settings[VIEWER])
+            if temporary_file is not None:
+                # NOTE: The idea is to sleep for WAIT_SECONDS minus the process runtime duration.
+                #       As a result, for a long-running process we don't wait more for nothing.
```
[the remark this answers is not in the document]
Comment (hartwork) [reply git-big-picture]: Done

File: test.scf (hunk 1 above)
[the remark this answers is not in the document]
Comment (esc) [reply test.scf:117]: Thank you for clarification, that was very helpful.

**Review by esc — APPROVED** [review]
Looks good, ready for merge.

**Pull request merged**